lib/config.js

```diff
@@ -701,42 +701,45 @@ util.loadFileConfigs = function() {
   });
 
   // Override configurations from the $NODE_CONFIG environment variable
-  var envConfig = {};
-  if (process.env.NODE_CONFIG) {
-    try {
-      envConfig = JSON.parse(process.env.NODE_CONFIG);
-    } catch(e) {
-      console.error('The $NODE_CONFIG environment variable is malformed JSON');
+  // NODE_CONFIG only applies to the base config
+  if (!configDir) {
+    var envConfig = {};
+    if (process.env.NODE_CONFIG) {
+      try {
+        envConfig = JSON.parse(process.env.NODE_CONFIG);
+      } catch(e) {
+        console.error('The $NODE_CONFIG environment variable is malformed JSON');
+      }
+      util.extendDeep(config, envConfig);
+      configSources.push({
+        name: "$NODE_CONFIG",
+        parsed: envConfig,
+      });
     }
-    util.extendDeep(config, envConfig);
-    configSources.push({
-      name: "$NODE_CONFIG",
-      parsed: envConfig,
-    });
-  }
 
-  // Override configurations from the --NODE_CONFIG command line
-  var cmdLineConfig = util.getCmdLineArg('NODE_CONFIG');
-  if (cmdLineConfig) {
-    try {
-      cmdLineConfig = JSON.parse(cmdLineConfig);
-    } catch(e) {
-      console.error('The --NODE_CONFIG={json} command line argument is malformed JSON');
+    // Override configurations from the --NODE_CONFIG command line
+    var cmdLineConfig = util.getCmdLineArg('NODE_CONFIG');
+    if (cmdLineConfig) {
+      try {
+        cmdLineConfig = JSON.parse(cmdLineConfig);
+      } catch(e) {
+        console.error('The --NODE_CONFIG={json} command line argument is malformed JSON');
+      }
+      util.extendDeep(config, cmdLineConfig);
+      configSources.push({
+        name: "--NODE_CONFIG argument",
+        parsed: cmdLineConfig,
+      });
     }
-    util.extendDeep(config, cmdLineConfig);
-    configSources.push({
-      name: "--NODE_CONFIG argument",
-      parsed: cmdLineConfig,
-    });
+
+    // Place the mixed NODE_CONFIG into the environment
+    env['NODE_CONFIG'] = JSON.stringify(util.extendDeep(envConfig, cmdLineConfig, {}));
   }
 
   // Override with environment variables if there is a custom-environment-variables.EXT mapping file
   var customEnvVars = util.getCustomEnvVars(CONFIG_DIR, extNames);
   util.extendDeep(config, customEnvVars);
 
-  // Place the mixed NODE_CONFIG into the environment
-  env['NODE_CONFIG'] = JSON.stringify(util.extendDeep(envConfig, cmdLineConfig, {}));
-
   // Extend the original config with the contents of runtime.json (backwards compatibility)
```

Comment:
I moved that up so that it could live in the same `if` block as the rest of the `NODE_CONFIG` support. I did not consider the case that the customEnvVars might be referring to NODE_CONFIG. Is that even possible, given that it is a JSON string? Did not look into that deeply yet.